libbeat/metric/system/process/process_aix.go

```diff
@@ -0,0 +1,190 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package process
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/user"
+	"strconv"
+	"syscall"
+	"unsafe"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/metric/system/resolve"
+	"github.com/elastic/beats/v7/libbeat/opt"
+)
+
+// FetchPids returns a map and array of pids
+func (procStats *Stats) FetchPids() (ProcsMap, []ProcState, error) {
+
+	info := C.struct_procsinfo64{}
+	pid := C.pid_t(0)
+
+	procMap := make(ProcsMap, 0)
+	var plist []ProcState
+	for {
+		// getprocs first argument is a void*
+		num, err := C.getprocs(unsafe.Pointer(&info), C.sizeof_struct_procsinfo64, nil, 0, &pid, 1)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "error fetching PIDs")
+		}
+		procMap, plist = procStats.pidIter(int(info.pi_pid), procMap, plist)
+
+		if num == 0 {
+			break
+		}
+	}
+	return procMap, plist, nil
+}
+
+// GetInfoForPid returns basic info for the process
+func GetInfoForPid(_ resolve.Resolver, pid int) (ProcState, error) {
+	info := C.struct_procsinfo64{}
+	cpid := C.pid_t(pid)
+
+	num, err := C.getprocs(unsafe.Pointer(&info), C.sizeof_struct_procsinfo64, nil, 0, &cpid, 1)
+	if err != nil {
+		return ProcState{}, errors.Wrap(err, "error in getprocs")
+	}
+	if num != 1 {
+		return ProcState{}, syscall.ESRCH
+	}
+
+	state := ProcState{}
+	state.Pid = opt.IntWith(pid)
+
+	state.Name = C.GoString(&info.pi_comm[0])
+	state.Ppid = opt.IntWith(int(info.pi_ppid))
+	state.Pgid = opt.IntWith(int(info.pi_pgrp))
+
+	switch info.pi_state {
```

Comment:
I have seen that you are setting the state with regular strings in several files. Would it be possible to refactor it to an `enum` or a `uint` type? This way we could avoid problems caused by typos.

Reply:
Ah, actually had that on my TODO list, and then forgot to check it off. Fixed.